packages/web-extension/src/messaging/BackgroundMessenger.ts

```diff
@@ -0,0 +1,80 @@
+// only tested in ../e2e tests
+import { ChannelName, Messenger, MessengerDependencies, MessengerPort, PortMessage } from './types';
+import { Logger } from 'ts-log';
+import { Subject, of } from 'rxjs';
+
+interface Channel {
+  message$: Subject<PortMessage>;
+  ports: Set<MessengerPort>;
+  hasMethodRequestHandler?: boolean;
+}
+
+/**
+ * Intended to be used in service worker background process.
+ * Manages connections with different parts of the extension.
+ * Connections are managed through ports.
+ * All other parts of extension are expected use NonBackgroundMessenger.
+ * You won't be able to add any additional 'runtime.onConnect' listeners in background process once this is called.
```

Comment:
Why so?

Reply:
It's something I found via trial and error. Multiple handlers in the same process just won't work.